[WIP] compatibility: Add links to app stores.

trueskawka opened the pull request:
Add links to app stores on the 'compatibility fail' screen.

The implemented link style is simple, it just underlines the store name. The current links are placeholders pointing to the Facebook app.

Tested on iOS 10.2 (iPad Mini, iPhone 6S) - the link redirects to the App Store. It's impossible to test it in the iOS Simulator, as there is no App Store available.

For testing, I've edited the `src/nav/NavigationContainer.js`:
- changed `compatibilityCheckFail` to false,
- commented out `checkCompatibility().then()`

borisyankov commented:
Move the actual URLs to config.js

borisyankov commented:
Maybe we want these types of buttons though:
![http://www.citclops.eu/media/citclops/wysiwyg/000160_appstore-button.png](http://www.citclops.eu/media/citclops/wysiwyg/000160_appstore-button.png)

What do you think?

trueskawka commented:
Cool, will move the URLs.

Yeah, I think buttons would be nicer :) Do we want to adjust text somehow, e.g. to "Download from the App Store", or just change the text before it, e.g. "Download new version <button>"?

borisyankov commented:
I guess it depends on how clear the buttons are. 
They seem self explanatory, so maybe just tell them that they have old version and the button lets them know they need to press to download.

trueskawka commented:
Added `png` buttons for both stores, moved the links to `config.js` and fixed linter errors - the build passed on my local instance.

Tested on iOS 10.2 (iPad Mini, iPhone 6S).

trueskawka commented on src/start/CompatibilityScreen.js:
```diff
@@ -15,17 +17,58 @@ const styles = StyleSheet.create({
-        <Text style={styles.text}>This app is too old!</Text>
-        <Text style={styles.text}>Please download the latest version from the App Store.</Text>
+        <Text style={styles.text}>This is an outdated version of Zulip.</Text>
+        {(Platform.OS === 'ios') ? (
```
I'm wondering if this if-else should be put somewhere else? 

Also, do we want to have a component for store link logos?

neerajwahi replied:
You can create a component called AppStoreLink and have two versions, one in `AppStoreLink.ios.js` and another in `AppStoreLink.android.js`. Then you just import it like `import AppStoreLink from './AppStoreLink'` and you don't have to worry about the platform check at all :)

React Native will automatically import the correct version for the platform

trueskawka replied:
Cool, will do, thanks!

neerajwahi commented on src/config.js:
```diff
@@ -4,4 +4,6 @@ export default {
   compatibilityUrl: 'https://zulipchat.com/compatibility',
   defaultLoginEmail: '', // useful during development, empty for production
   defaultLoginPassword: '', // useful during development, empty for production
+  appStoreLink: 'https://itunes.apple.com/us/app/facebook/id284882215?mt=8',
```
I assume these are placeholder links?

Probably best to set these to the empty string before checking in this diff

trueskawka replied:
Yeah, those are placeholders for testing purposes - will be sure to remove them after final review :)

neerajwahi commented on src/start/CompatibilityScreen.js:
```diff
@@ -15,17 +17,58 @@ const styles = StyleSheet.create({
+  handleClick = (link: string) => {
+    Linking.canOpenURL(link).then(supported => {
+      if (!supported) {
+        console.log('Can\'t open URL: ', link);
```
Maybe show a modal alert here that says something like 'Could not open app store.'

trueskawka replied:
Good point, will do!

trueskawka commented:
Created a component for `AppStoreLink`, added an alert.

Placeholder links are still in place for testing purposes - will remove them after final review.

neerajwahi commented on src/start/AppStoreLink.android.js:
```diff
@@ -0,0 +1,33 @@
+import React from 'react';
```
Hmm.. it actually looks like the only difference between `AppStoreLink.android.js` and `AppStoreLink.ios.js` is the image source, so maybe it's better if we merge these into one file (`AppStoreLink.js`) and choose the image source based on the platform.

I like having this as a separate component though!

neerajwahi commented on src/start/CompatibilityScreen.js:
```diff
@@ -15,17 +16,25 @@ const styles = StyleSheet.create({
+  handleClick = (link: string) => {
+    Linking.canOpenURL(link).then(supported => {
+      if (!supported) {
+        Alert.alert('Loading error', 'Could not open the store.');
```
Maybe change this to 'Please go to the app store to download the latest version'?

neerajwahi commented on src/start/CompatibilityScreen.js:
```diff
@@ -15,17 +16,25 @@ const styles = StyleSheet.create({
 export default class CompatibilityScreen extends React.Component {
-
+  handleClick = (link: string) => {
+    Linking.canOpenURL(link).then(supported => {
```
It would be nice to use async/await here! It would look like:
```
try {
  const supported = await Linking.canOpenURL(link);
  if (!supported) {
    ...
  } else {
    ...
  }
} catch (e) {
  ...
}
```

You'll also have to mark `handleClick` as async function

neerajwahi commented on src/start/CompatibilityScreen.js:
```diff
@@ -15,17 +16,25 @@ const styles = StyleSheet.create({
         <Logo />
-        <Text style={styles.text}>This app is too old!</Text>
-        <Text style={styles.text}>Please download the latest version from the App Store.</Text>
+        <Text style={styles.text}>This is an outdated version of Zulip.</Text>
```
Nice improvement! I'd keep the 'Please download the latest version from the App Store:' text though so it's more of a call to action

neerajwahi commented:
@trueskawka just checking in to see how things are going with this PR. Let me know if you have any questions or I can be of any help

borisyankov commented:
@trueskawka a lot of the code looks good.
On the other hand, a lot of code needs to be changed in subtle ways.
When you have time to work on it, please reopen and we can finish it up 👍
